compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELinkExecutable.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/Transforms/PassDetail.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "iree/compiler/Codegen/Utils/LinkingUtils.h"
+#include "iree/compiler/Utils/ModuleUtils.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+struct AMDAIELinkExecutablesPass
+    : public impl::AMDAIELinkExecutablesBase<AMDAIELinkExecutablesPass> {
+  AMDAIELinkExecutablesPass() = default;
+  void runOnOperation() override {
+    auto moduleOp = getOperation();
+    auto moduleBuilder = OpBuilder::atBlockBegin(moduleOp.getBody());
+
+    auto sourceExecutableOps =
+        llvm::to_vector<8>(moduleOp.getOps<IREE::HAL::ExecutableOp>());
```

Comment:
Its not limited to 8, its a performance hint for the compiler to allocate space for a vector of 8, but the actual content can be more or less. I think using powers of 2 is common here and since typical program would have several executables we use 8 here. Btw I copied most of this pass from this upstream pass (which had the 8)
https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULinkExecutables.cpp#L26